### DISCUSSION TITLE
Random pauses (and occasional disconnects)

### DISCUSSION THREAD

**Discussion by omaristalis** [dotnet/yarp#446] [General] [answered]
Just wanted to see if anyone else has seen what I'm seeing while testing YARP using SignalR.  Apologies, as this is somewhat of a ramble, and there's probably two separate problems here.

My application is pushing data back through a SignalR websocket from the server to the client (connection.On() at the client side).  Two connections are running in my test, each pushing around 50Mbit/sec

Everything seems fine, but I see occasional pauses in sending - from a second to multiple seconds.  The service is still pushing data, because when the freeze ends the network usage spikes and all of that data makes it through.

When I connect direct to my service, I don't see pauses at all.

The disconnects are more concerning, however.  I don't see anything in the output (although I should turn on more detailed logging), but twice yesterday my clients suddenly stopped receiving data altogether, from which they never recovered.  I think I remember an exception being generated somewhere eventually, but I'd need to re-test to work out where.

Any comments welcome!

**Comment by Tratcher** [comment, marked as answer]
Any chance it was a GC pause? There should be runtime events for that.

As we work on perf optimizations and reducing allocations we should be able to limit the effects of GC.

Please open a separate issue for the disconnects when you've got more data.

@MihaZupan this is a good troubleshooting exercise to consider for metrics/logging.

**Comment by omaristalis** [comment]
Sounds good - thanks for the reply.

I'll do some more testing around the disconnects.
